src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java

```diff
@@ -93,6 +93,10 @@ public PersonalSchedule postSchedule(
 
         PersonalSchedule personalschedule = new PersonalSchedule();
         personalschedule.setUser(currentUser.getUser());
+        int maxLength = 15;
```

Comment:
A few things...

So, the first problem is that we are getting a syntax error with the reference to name in the lambda:

```
local variables referenced from a lambda expression must be final or effectively final
```


But if you make it final by changing 
```
@ApiParam("name") @RequestParam String name, 
```

to 
```
@ApiParam("name") @RequestParam final String name,
```
which you can do... Then you hav a problem with reassigning name here:
```
      name = name.substring(0, maxLength);
```

That problem can be fixed by adding a separate variable, something like "nameToAssign" perhaps

Finally, though it may not have been clear from the original issue, we also need this logic in the two PUT routes that can update name (which will be used later for the update function).

So, I'd suggest hoisting this variable:
```
    int maxLength = 15; 
```
and making it a class level constant, maybe something like:

```
static final int MAX_NAME_LENGTH=15;
```
so you can use it in both the POST and PUT routines.

Reply:
After looking at Vicki's PR that was just merged, I am thinking it is better to have error checking similar to hers in the backend (that is: throwing illegalArgumentException for arguments which are too long)

It makes more sense to me that we'd be getting consistent results in the frontend and backend for the same input.